cmd/kubelet/app/options/options.go

```diff
@@ -64,6 +60,20 @@ type KubeletServer struct {
 	// If runOnce is true, the Kubelet will check the API server once for pods,
 	// run those in addition to the pods specified by the local manifest, and exit.
 	RunOnce bool
+
+	// HostnameOverride is the hostname used to identify the kubelet instead
+	// of the actual hostname.
+	HostnameOverride string
+	// NodeIP is IP address of the node.
+	// If set, kubelet will use this IP address for the node.
+	NodeIP string
+}
```

Comment:
I think the `KubeletFlag` type can have its own `AddFlags()` method, which `KubeletServer` can call. Thoughts?